io-engine/src/bdev/nexus/nexus_bdev_snapshot.rs

```diff
@@ -191,19 +180,65 @@ impl ReplicaSnapshotExecutor {
-                    ?snapshot_params,
-                    "Starting nexus replica snapshot operation",
-                );
+                let nexus_name = self.nexus_name.clone();
+
+                // Schedule replica snapshot operation on master core.
+                let rx = Reactor::spawn_at_primary(async move {
+                    debug!(
+                        replica_uuid,
+                        ?snapshot_params,
+                        "Starting nexus replica snapshot operation",
+                    );
+
+                    let replica = nexus_lookup(&nexus_name)
+                        .ok_or_else(|| {
+                            error!(
+                                nexus_name,
+                                replica_uuid,
+                                "Failed to lookup nexus device, skipping snapshot for replica"
+                            );
+                            CoreError::BdevNotFound {
+                                name: nexus_name.clone()
+                            }
+                        })
+                        .and_then(|n| {
+                            n.child_by_uuid(&replica_uuid).map_err(|error| {
+                                error!(
+                                    nexus_name,
+                                    replica_uuid,
+                                    ?error,
+                                    "Failed to lookup replica device, skipping snapshot for replica"
+                                );
+                                CoreError::BdevNotFound {
+                                    name: replica_uuid.clone(),
+                                }
+                            })
+                        })?;
+
+                    let handle = replica
+                        .get_io_handle_nonblock()
+                        .await
+                        .map_err(|error| {
+                            error!(
+                                nexus_name,
+                                replica_uuid,
+                                ?error,
+                                "Failed to get I/O handle for replica device, skipping snapshot for replica"
+                            );
+                            CoreError::GetIoChannel { name: replica_uuid.clone() }
+                        })?;
+
+                    handle.create_snapshot(snapshot_params).await
+                })
+                .expect("Can't schedule replica snapshot operation");
 
-                // Snapshot operation future shall be able to track back to the
-                // replica.
+                // Snapshot operation future shall be able to track back to the replica.
                 async move {
                     (
-                        replica_uuid,
-                        ctx.handle.create_snapshot(snapshot_params).await,
+                        ctx.replica_uuid.clone(),
+                        rx.await.expect("Snapshot sender disappeared"),
```

Comment:
this something that we might need to handle, if futures are dropped (eg: during shutdown), I found something similar during keep alive testing on device_monitor which also uses spawn_at_primary.